File: solutions/cpp/0605-can-place-flowers.cpp
class Solution {
public:
    bool canPlaceFlowers(std::vector<int>& flowerbed, int n) {
        flowerbed.insert(flowerbed.begin(), 0);
        flowerbed.push_back(0);
        for(int i = 1; i < flowerbed.size() - 1; ++i) {
            if(flowerbed[i-1] == 0 && flowerbed[i] == 0 && flowerbed[i+1] == 0) {
                flowerbed[i] = 1;
                --n;
            }
            if(n == 0)
                return true;
        }
        return n <= 0;
    }
};

File: solutions/csharp/0605-can-place-flowers.cs
public class Solution {
    public bool CanPlaceFlowers(int[] flowerbed, int n) {
        int[] bed = new int[flowerbed.Length + 2];
        Array.Copy(flowerbed, 0, bed, 1, flowerbed.Length);
        for (int i = 1; i < bed.Length - 1; ++i) {
            if (bed[i - 1] == 0 && bed[i] == 0 && bed[i + 1] == 0) {
                bed[i] = 1;
                --n;
            }
            if (n == 0)
                return true;
        }
        return n <= 0;
    }
}
